src/forms/FormFieldNumberTable/FormFieldNumberTable.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { FieldDef } from "@root/components/Field/FieldTypes";
+import { ButtonProps } from "@root/components/Button";
+import { columns, numberTableDefaultValue, rows } from "./numberTableUtils";
+
+const NumberTableExample = (): ReactElement => {
+	const { state, dispatch } = useForm();
+
+	const onSubmit = async () => {
+		const { valid, data } = await dispatch(formActions.submitForm());
+		if (!valid) return;
+
+		alert("Form submitted with the following data: " + JSON.stringify(data, null, " "));
+	};
+
+	const fields = useMemo(
+		(): FieldDef[] => [
+			{
+				name: "numberTable",
+				label: "Label",
+				type: "numberTable",
+				defaultValue: numberTableDefaultValue,
+				inputSettings: {
+					rowTotalLabel: "TOTAL",
+					columnTotalLabel: "No. Rooms",
+					topLeftLabel: "Day",
+					rows: rows,
+					columns: columns
+				},
+			},
+		],
+		[
+			numberTableDefaultValue,
+			rows,
+			columns
+		]
+	);
+
+	const buttons: ButtonProps[] = [
+		{
+			label: "Save",
+			onClick: onSubmit,
+			color: "yellow",
+			variant: "contained",
+		},
+	];
+
+	return (
+		<Form
+			buttons={buttons}
+			title="Form Title"
+			description="This is a description example"
+			state={state}
+			fields={fields}
+			dispatch={dispatch}
+		/>
+	);
+};
+
+
+const mockResizeObserver = jest.fn();
+mockResizeObserver.mockReturnValue({
+	observe: () => null,
+	unobserve: () => null,
+	disconnect: () => null
+});
+window.ResizeObserver = mockResizeObserver;
+const scrollIntoViewMock = jest.fn();
+window.HTMLElement.prototype.scrollTo = scrollIntoViewMock;
+
+
+afterEach(cleanup);
+
+describe("FormFieldMatrix component", () => {
+	beforeEach(async () => {
+		await act(() => {
+			render(
+				<NumberTableExample/>
+			)
+		});
+	});
+
+	it("should display the totals and top left labels", () => {
+		expect(screen.getByText("TOTAL")).toBeInTheDocument();
+		expect(screen.getByText("No. Rooms")).toBeInTheDocument();
+		expect(screen.getByText("TOTAL")).toBeInTheDocument();
+		expect(screen.getByText("Day")).toBeInTheDocument();
+	});
+
+	it("should display the columns titles", () => {
+		expect(screen.getByText("Single")).toBeInTheDocument();
+		expect(screen.getByText("Double")).toBeInTheDocument();
+		expect(screen.getByText("Queen")).toBeInTheDocument();
+		expect(screen.getByText("King")).toBeInTheDocument();
+		expect(screen.getByText("Suite")).toBeInTheDocument();
+		expect(screen.getByText("Any")).toBeInTheDocument();
+	});
+
+	it("should display the columns titles", () => {
+		expect(screen.getByText("Single")).toBeInTheDocument();
+		expect(screen.getByText("Double")).toBeInTheDocument();
+		expect(screen.getByText("Queen")).toBeInTheDocument();
+		expect(screen.getByText("King")).toBeInTheDocument();
+		expect(screen.getByText("Suite")).toBeInTheDocument();
+		expect(screen.getByText("Any")).toBeInTheDocument();
+	});
+
+	it("should display 24 fields since the table size is 6x4", () => {
+		const inputs = screen.getAllByPlaceholderText("0");
```

Comment:
Tests-ids are added, but the objective of this test is to show that the table contains a total o 24 input fields. Tests id would not be helpful in this scenario, since each test-id is unique per field.